nrf52840-hal/src/gpio.rs

```diff
@@ -0,0 +1,365 @@
+//! General Purpose Input / Output
```

Comment:
this file looks similar to `nrf52-hal/src/gpio.rs`... is it close enough that you could copy that over this one and maybe tweak the `use nrf52;` lines to `use nrf52840`?

Reply:
That's because it was originally a copy of an earlier version. I've updated it and made a couple of tweaks for the 52840. Couldn't get the GPIOs on P1 to work with the macro though due to the shared module p0, and I'm not sure of the best way to fix it.